src/vcpkg/sourceparagraph.cpp

```diff
@@ -510,10 +510,15 @@ namespace vcpkg
             return t;
         }
 
+        static std::string map_illegal_names(std::string&& name)
+        {
+            return Json::IdentifierDeserializer::is_ident(name) ? std::move(name) : "<error>";
+        }
+
         Optional<DependencyRequestedFeature> visit_string(Json::Reader& r, StringView sv) const override
         {
             return Json::IdentifierDeserializer::instance.visit_string(r, sv).map(
-                [](std::string&& name) { return std::move(name); });
+                [](std::string&& name) { return map_illegal_names(std::move(name)); });
         }
```

Comment:
This should work exactly like `Dependency` below:

```suggestion
        Optional<DependencyRequestedFeature> visit_string(Json::Reader& r, StringView sv) const override
        {
            return Json::IdentifierDeserializer::instance.visit_string(r, sv).map(
                [&r](std::string&& name) {
                  if (!Json::IdentifierDeserializer::is_ident(name))
                    r.add_generic_error(type_name(), msg::format(msgInvalidDependency));
                  return std::move(name);
                });
        }
```

Reply:
This code would still result in a crash since the constructor of `DependencyRequestedFeature` `check_exit` that the feature name is not "core" or "default" because there is a member `default-features` that must be used.

Reply:
> This code would still result in a crash since the constructor of `DependencyRequestedFeature` `check_exit` that the feature name is not "core" or "default" because there is a member `default-features` that must be used.

The point is that things that are deserialized can't have check_exits like that.

Notably, the assert doesn't even succeed in maintaining the invariant that it won't be empty, since:

```
DependencyRequestedFeature f{"nonempty"};
auto moved = std::move(f);
assert(f.name.empty()); // passes on most standard library implementations
```